File: cases/taint/java/local-chain-positive/LocalChainPositive.java
package dataflowbench.taint;

final class LocalChainPositive {
    static int dfb_source() { // DFB-SOURCE: local-chain-input
        return 1;
    }

    static void dfb_sink(int value) { } // DFB-SINK: local-chain-sink

    static void run() {
        int first = dfb_source();
        int second = first; // DFB-WITNESS: local-chain-second
        int third = second; // DFB-WITNESS: local-chain-third
        dfb_sink(third);
    }
}

File: cases/taint/javascript/local-chain-positive/LocalChainPositive.js
function dfb_source() { // DFB-SOURCE: local-chain-input
  return "tainted";
}

function dfb_sink(value) {} // DFB-SINK: local-chain-sink

function run() {
  const first = dfb_source();
  const second = first; // DFB-WITNESS: local-chain-second
  const third = second; // DFB-WITNESS: local-chain-third
  dfb_sink(third);
}
